### DISCUSSION TITLE
linguist-generated=true vs. sloc count

### DISCUSSION THREAD

**Discussion by szepeviktor** [github-linguist/linguist#6595] [Classification] [answered]
Hello! We have just added `linguist-generated=true` to our generated files.
https://github.com/szepeviktor/unipoints/blob/main/.gitattributes
There are 146k non-empty lines `git ls-files -- '*.php' | xargs -- cat | sed -e '/\S/!d' | wc -l`
Non-generated code amounts to 1500 non-empty lines.

Linguist says
```
99.97%  61531      PHP
0.03%   20         Batchfile
```
https://github.com/szepeviktor/linguist-on-you/actions/runs/6682473134#summary-18157685264

How can we make linguist ignore generated code and count only 1500 lines?

```
github.com/AlDanial/cloc
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
PHP                            357          72131          72429          74422
-------------------------------------------------------------------------------
```
after removing generated files
```
github.com/AlDanial/cloc
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
PHP                             26            275            243           1259
-------------------------------------------------------------------------------
```

Thank you.

**Comment by lildude** [comment, marked as answer]
I'm not sure why you'd want to do this as Linguist doesn't use LOC when determining the language proportions, it uses bytes of code (the second column in your first output).

From [How Linguist works](https://github.com/github-linguist/linguist/blob/master/docs/how-linguist-works.md#how-linguist-works):

> The percentages are calculated based on the bytes of code for each language as reported by the [List Languages](https://docs.github.com/rest/reference/repos#list-repository-languages) API.

You can see Linguist is ignoring those files you've marked as generated if you use the `--breakdown` flag:

```console
➜ github-linguist --breakdown ./unipoints/
99.97%  61531      PHP
0.03%   20         Batchfile

PHP:
.php-cs-fixer.php
bin/unipoints
build/BlocksBuilder.php
build/BlocksBuilder/Block.php
build/CodepointsBuilder.php
build/CodepointsBuilder/Codepoint.php
build/CodepointsBuilder/Name.php
build/CodepointsBuilder/NameAlias.php
build/CodepointsBuilder/NameAlias/Type.php
build/CodepointsBuilder/UnicodeData.php
build/DataStorage.php
build/FilesystemCache.php
build/UserMessageException.php
src/Category.php
src/Info/BlockInfo.php
src/Info/CategoryInfo.php
src/Info/CodepointInfo.php
src/Info/PlaneInfo.php
src/Info/UnicodeInfo.php
src/Plane.php
test/bootstrap.php
test/service/BlocksBuilderWrapper.php
test/tests/BlocksBuilderTest.php
test/tests/CategoryEnumTest.php
test/tests/DataStorageTest.php
test/tests/FilesystemCacheTest.php
test/tests/PlaneEnumTest.php

Batchfile:
bin/unipoints.bat

➜ 
```

If we count all those PHP file we get the 26 files in your last output block:

```console
➜ github-linguist --breakdown ./unipoints/ | grep '.php' | wc -l
      26
➜ 
```

**Reply by szepeviktor** [reply]
Thank you very much.
Very kind of you.

Now I know that number is a byte counter!
